app/models/energy/programtype.py

```diff
@@ -6,6 +6,7 @@
 from datetime import datetime
 from app.models.energy.scheduleruleset import ScheduleRulesetAbridged
 from app.models.energy.schedulefixedinterval import ScheduleFixedIntervalAbridged
+import re
```

Comment:
Are we using `re` in this module?  It looks like we are not and we should probably remove it.